django/utils/functional.py

```diff
@@ -128,6 +128,19 @@ def __cast(self):
             else:
                 return func(*self.__args, **self.__kw)
 
+        if six.PY3:
+            def __str__(self):
+                return six.text_type(self.__cast())
```

Comment:
Maybe it's better to use `str` here instead of `six.text_type` since you already know that you are in Py3 at this point.
Same for the other occurrences of `six.*_type`.

Or just use `python_2_unicode_compatible` instead.